src/com/google/enterprise/adaptor/MetadataTransform.java

```diff
@@ -35,6 +36,41 @@
   public static final String KEY_TRANSMISSION_DECISION 
       = "Transmission-Decision";
 
+  /**
+   * Identifies the location a configured key/value pair resides,
+   * either in the {@link Metadata}, or the map of {@code params}.
+   * <p>
+   * Suppose a transform wishes examine or modify the display URL, which is
+   * stored in the {@code params}. The transform's configuration might look
+   * like:
+   * <pre><code>
+     key=Display-URL
+     keyset=params
+     </code></pre>
+   * If unspecified, the default value is {@code metadata}.
+   *
+   * @since 4.1.4
+   */
+  public enum Keyset {
+    METADATA("metadata"),
+    PARAMS("params");
+
+    private final String name;
+
+    private Keyset(String name) {
+      this.name = name;
+    }
+
+    public static Keyset from(String val) {
```

Comment:
Add javadoc and a throws tag for IllegalArgumentException, and do the same for Transmission-Decision. The enum values could stand some javadoc, too. There is no mention in the javadoc of what the string names for the enum values actually are. Don't forget that you can add javadoc to the individual values of the enum as well (e.g., METADATA and PARAMS).